store/mongo/datastore_mongo.go

```diff
@@ -209,54 +209,26 @@ func (db *DataStoreMongo) GetDevices(ctx context.Context, q store.ListQuery) ([]
 		if !q.Sort.Ascending {
 			sortFieldQuery[sortField] = -1
 		}
-		sortQuery = bson.M{"$sort": sortFieldQuery}
-	}
-	limitQuery := bson.M{"$skip": 0}
-	// exchange the limit query only if limit is set, as limits need to be positive in an aggregation pipeline
-	if q.Limit > 0 {
-		limitQuery = bson.M{"$limit": q.Limit}
-	}
-	combinedQuery := bson.M{
-		"$facet": bson.M{
-			"results": []bson.M{
-				sortQuery,
-				{"$skip": q.Skip},
-				limitQuery,
-			},
-			"totalCount": []bson.M{
-				{"$count": "count"},
-			},
-		},
-	}
-	resultMap := bson.M{
-		"$project": bson.M{
-			"results": 1,
-			"totalCount": bson.M{
-				"$ifNull": []interface{}{
-					bson.M{
-						"$arrayElemAt": []interface{}{"$totalCount.count", 0},
-					},
-					0,
-				},
-			},
-		},
+		findOptions.SetSort(sortFieldQuery)
 	}
 
-	cursor, err := c.Aggregate(ctx, []bson.M{
-		filter,
-		combinedQuery,
-		resultMap,
-	})
+	cursor, err := c.Find(ctx, findQuery, findOptions)
+	if err != nil {
+		return nil, -1, errors.Wrap(err, "failed to search devices")
+	}
 	defer cursor.Close(ctx)
 
-	if !cursor.Next(ctx) {
-		return nil, 0, nil
+	devices := []model.Device{}
+	if err = cursor.All(ctx, &devices); err != nil {
+		return nil, -1, errors.Wrap(err, "failed to search devices")
 	}
-	res := internalDeviceResult{}
```

Comment:
You can also delete this type definition (`internalDeviceResult`) as it's only used here.

Reply:
Right, done!